src/Aspire.Dashboard/Components/Pages/TraceDetail.razor.cs

```diff
@@ -93,6 +94,43 @@ private ValueTask<GridItemsProviderResult<SpanWaterfallViewModel>> GetData(GridI
             Items = page.ToList(),
             TotalItemCount = visibleSpanWaterfallViewModels.Count
         });
+
+        // Return spans where any child span matches the filter.
+        List<SpanWaterfallViewModel> GetFilteredSpanWaterfalls(List<SpanWaterfallViewModel> viewModels)
+        {
+            var viewModelsToFilterMatch = new Dictionary<SpanWaterfallViewModel, bool>();
```

Comment:
Is this a perf optimization to avoid filtering a span multiple times when there are children? Could you add a comment.